tests/macsec/test_neighbor_reboot.py

```diff
@@ -0,0 +1,41 @@
+'''
+
+The test case will verify a neighbor reboot while Macsec is Active.
+
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on neighbor and reboot
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+from time import sleep
+import pytest
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology("t2")
+]
+
+
+def test_neighbor_reboot(duthost, nbrhosts, request, enum_rand_one_frontend_asic_index):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Neighbor type must be sonic")
+    asic_index = enum_rand_one_frontend_asic_index
+    namespace = duthost.get_namespace_from_asic_id(asic_index)
+    dut_lldp_table = duthost.shell("show lldp table")['stdout'].split("\n")[3].split()
+    dut_to_neigh_int = dut_lldp_table[0]
+    neighhost = nbrhosts[dut_lldp_table[1]]["host"]
+    neighhost.shell("config save -y")
```

Comment:
Why do we need to save config here? There is no config change in the test function.

Reply:
This is standard practice for our tests before performing a reboot to ensure any changes are saved.